Composer/packages/client/src/components/ProjectTree/ProjectTree.tsx

```diff
@@ -239,7 +239,9 @@ export const ProjectTree: React.FC<Props> = ({
   }));
 
   useEffect(() => {
-    setSelectedLink(defaultSelected);
+    if (!isEqual(selectedLink, defaultSelected)) {
+      setSelectedLink(defaultSelected);
+    }
```

Comment:
What scenario causes this effect to run? When does `defaultSelected` change any why?

Reply:
@beyackle could you assist here

Reply:
HI, @a-b-r-o-w-n Thanks to review the code. The trigger flow is: update dialog data -> update the dialogs in the recoil -> update the current dialog in the <DesignPage/> -> update the trigger item -> update the defaultSelected.

And I will sync with @beyackle if we can remove the useEffect directly, or why we need it here.

Reply:
it's probably from when it parses the url and knows what should be selected in the tree based on url params.

Reply:
Thanks @hatpick , I have sync with @beyackle and @srinaath . We need this for changes like add a new dialog, a new trigger at the design page level. And we can make it more efficient by breaking the object into fields.